tests/integration/suite/test_workloads.py

```diff
@@ -310,6 +312,43 @@ def test_statefulset_workload_volumemount_subpath(admin_pc):
         assert e.value.error.status == 422
 
 
+def test_workload_redeploy(admin_pc, remove_resource):
+    client = admin_pc.client
+    ns = admin_pc.cluster.client.create_namespace(
+        name=random_str(),
+        projectId=admin_pc.project.id)
+    remove_resource(ns)
+    name = random_str()
+    timestamp = datetime.datetime.now().isoformat()
+
+    workload = client.create_workload(
+        name=name,
+        namespaceId=ns.id,
+        scale=1,
+        containers=[{
+            'name': 'one',
+            'image': 'nginx',
+        }],
+        annotations={
+            "cattle.io/timestamp":
+                timestamp}, )
+    remove_resource(workload)
+
+    client.action(workload, 'redeploy')
+
+    def _timestamp_reset():
+        workloads = client.list_workload(uuid=workload.uuid).data
+        return len(workloads) > 0 and workloads[0].annotations[
+            'cattle.io/timestamp'] != timestamp
+
+    wait_for(_timestamp_reset,
+             fail_handler=lambda: 'Timed out waiting for timestamp reset')
+
+    new_timestamp = client.list_workload(
+        uuid=workload.uuid).data[0].annotations['cattle.io/timestamp']
+    assert new_timestamp != timestamp
```

Comment:
nit: you are pretty much already asserting this since if it is not true then timestamp_reset will timeout and your fail_handler message will be shown.